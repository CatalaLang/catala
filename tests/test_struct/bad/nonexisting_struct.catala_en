## Article

```catala
declaration structure Foo:
  data f content integer

declaration scope A:
  context x content Foo
  context y content integer

scope A:
  definition x equals Foo { -- f: 1 }
  definition y equals x.Fo.f
```

```catala-test-inline
$ catala Interpret -s A
[ERROR]
No struct named Fo found

┌─⯈ tests/test_struct/bad/nonexisting_struct.catala_en:13.25-13.27:
└──┐
13 │   definition y equals x.Fo.f
   │                         ‾‾
   └─ Article
#return code 123#
```
