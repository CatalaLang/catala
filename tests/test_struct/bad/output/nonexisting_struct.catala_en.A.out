[ERROR] Struct Fo has not been defined before
[ERROR] 
[ERROR]   --> test_struct/bad/nonexisting_struct.catala_en
[ERROR]    | 
[ERROR] 13 |   definition y equals x.Fo.f
[ERROR]    |                         ^^
[ERROR]    + Article
